test/integration/modules/edgecore/eventbus/eventbus_test.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"database/sql"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/test/integration/utils/common"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	_ "github.com/mattn/go-sqlite3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//DeviceUpdate device update
+type DeviceUpdate struct {
+	State string `json:"state,omitempty"`
+}
+
+//the struct of device
+type Device struct {
+	id          []uint8
```

Comment:
why is the device id an array ?

Reply:
Created the device structure based in schema used by edge_core for creating the device table. like shown below

  CREATE TABLE IF NOT EXISTS `device` (
        `id` varchar(64) NOT NULL PRIMARY KEY,
        `name` text,
        `description` text,
        `state` text,
        `last_online` text
    );

But, string type is also able to decode from the device query, now structure is re-defined with "string" type for all variables.